xdsl/tools/xdsl_tblgen.py

```diff
@@ -330,7 +330,7 @@ def _resolve_prop_constraint(self, rec: TblgenRecord | str) -> str:
             case "StrAttr":
                 return "BaseAttr(StringAttr)"
             case "SymbolNameAttr":
```

Comment:
why keep this?

Reply:
https://github.com/llvm/llvm-project/blob/299a55a88fae4fc423c440436b2632d2a6bd800a/mlir/include/mlir/IR/CommonAttrConstraints.td#L348 relevant definition

Reply:
I think this is exactly the pattern you were talking about in the other PR:
[op](https://github.com/llvm/llvm-project/blob/main/mlir/test/lib/Dialect/Test/TestOpsSyntax.td#L193-L197),
[test](https://github.com/llvm/llvm-project/blob/main/mlir/test/mlir-tblgen/op-format.mlir#L29-L31).

Reply:
My feeling is that if we can figure out `SymbolNameAttr` then adding a new public method to `Printer` shouldn't be necessary

Reply:
Tablegen expert hat on:
`SymbolNameAttr` is literally just a `StringAttr`, but that gets printed as a symbol (like `@name`).
I think it also adds a method as well, but I may be wrong on that.